src/Analyzer.JsonRuleEngine.UnitTests/RegexOperatorTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.UnitTests
+{
+    [TestClass]
+    public class RegexOperatorTests
+    {
+        [DataTestMethod]
+        [DataRow("value", "aaavalueaaa", DisplayName = "String contains \"value\"")]
+        [DataRow("value", "aaaVaLuEaaa", DisplayName = "String contains \"value\" - case-insensitive")]
+        [DataRow("^value", "valueaaa", DisplayName = "String begins with \"value\"")]
+        [DataRow("value$", "aaavalue", DisplayName = "String ends with \"value\"")]
+        [DataRow("^((?!value).)*$", "aaa", DisplayName = "String does not contain \"value\"")]
+        public void EvaluateExpression_StringMatchesRegex_EvaluationIsTrue(string regex, string stringToMatch)
+        {
+            // {"Regex": jTokenValue} is true
+            var regexOperator = new RegexOperator(regex, isNegative: false);
+            Assert.IsTrue(regexOperator.EvaluateExpression(stringToMatch));
+        }
+
+        [DataTestMethod]
+        [DataRow("value", "aaa", DisplayName = "String does not contain \"value\"")]
+        [DataRow("^value", "aaavalue", DisplayName = "String does not begin with \"value\"")]
+        [DataRow("value$", "valueaaa", DisplayName = "String does not end with \"value\"")]
+        [DataRow("^((?!value).)*$", "value", DisplayName = "String contains \"value\"")]
+        public void EvaluateExpression_StringDoesNotMatchRegex_EvaluationIsFalse(string regex, string stringToMatch)
+        {
+            // {"Regex": jTokenValue} is false
+            var regexOperator = new RegexOperator(regex, isNegative: false);
+            Assert.IsFalse(regexOperator.EvaluateExpression(stringToMatch));
+        }
+
```

Comment:
Add a test with the `regex` operator having an invalid regular expression (like a character group or match group not having a closing bracket, or escaping an invalid character, or something).

Reply:
What is the expected behavior here? I assume we'd want to just throw the exception so they can fix the rule right?